non-oss/pharos_pro/phases/migrate_master.rb

```diff
@@ -1,12 +1,40 @@
 # frozen_string_literal: true
 
+require 'pharos/phases/mixins/cluster_version'
+
 module Pharos
   module Phases
     class MigrateMaster < Pharos::Phase
       title "Migrate master"
 
+      include Pharos::Phases::Mixins::ClusterVersion
+
       def call
-        logger.info { 'Nothing to migrate.' }
+        if existing_version == pharos_version
+          logger.info 'Nothing to migrate.'
+        elsif existing_version > build_version('1.0.0') && existing_version < build_version('2.3.0-alpha.1')
+          logger.info 'Migrating cluster info ...'
+          migrate_cluster_info
+        end
+      end
+
+      def migrate_cluster_info
+        cm_client = kube_client.api('v1').resource('configmaps', namespace: 'kube-public')
+        cluster_info = cm_client.get('cluster-info')
+        cm_client.delete('cluster-info')
+
+        new_cluster_info = K8s::Resource.new(
+          kind: cluster_info.kind,
+          apiVersion: cluster_info.apiVersion,
+          metadata: {
+            name: cluster_info.metadata.name,
+            namespace: cluster_info.metadata.namespace
+          },
+          data: cluster_info.data.to_h
+        )
+        cm_client.create_resource(new_cluster_info)
+      rescue K8s::Error::NotFound
+        logger.error { "cluster-info configmap has gone, skipping the migration ..." }
```

Comment:
Where has it gone? If this is just a check if something needs to be done, does this need to be an error?

Reply:
Good point, it could be `info` level message.

Reply:
Though not sure how well whole cluster functions if cluster-info has gone awol 🤔

Reply:
Also, logger block format messages only make sense for `debug` and even then if the message contains interpolation :)

Reply:
> Also, logger block format messages only make sense for `debug` and even then if the message contains interpolation :)

![old-habit-die-hard](https://media.giphy.com/media/rx2QtLHoW7D32/giphy.gif)